Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -248,8 +260,9 @@ void MarkupAccumulator::serializeNodesWithNamespaces(Node& targetNode, Serialize
                 }
             }
 
+            bool shouldSkipChidren = appendContentsForNode(m_markup, *current);
```

Comment:
It's not great that we're making the traversal logic more complicated but I don't have a better suggestion.